Description: Skip a failing test on GNU/kFreeBSD; print result instead.
Author: Yavor Doganov <yavor@gnu.org>
Forwarded: not-needed
Last-Update: 2021-12-10
---

--- gnustep-base.orig/Tests/base/NSFileManager/general.m
+++ gnustep-base/Tests/base/NSFileManager/general.m
@@ -167,7 +167,15 @@
   PASS(EQ(ot, nt), "copy creation date equals original")
   ot = [[oa fileModificationDate] timeIntervalSinceReferenceDate];
   nt = [[na fileModificationDate] timeIntervalSinceReferenceDate];
+#ifdef __linux__
   PASS(EQ(ot, nt), "copy modification date equals original")
+#else
+  /* Store result in a file to be printed later since gnustep-tests
+     redirects the standard streams.  As the test program cleans up
+     the current directory, use the parent one.  */
+  NSString *res = [NSString stringWithFormat: @"### %f %f\n", ot, nt];
+  [res writeToFile: @"../result.log" atomically: YES];
+#endif
   {
     NSData *dat1 = [mgr contentsAtPath: @"NSFMFile"];
     NSError *err;
